Polyhedra/PolyhedraFullLayer.swift

```diff
@@ -1,50 +1,58 @@
 import AppKit
 
 class PolyhedraFullLayer: CALayer {
-    private let polyhedraLayer: PolyhedraLayer
-    private let cachedRenderings: [CachedRendering]
-    private let radius: CGFloat
-    private let maxX: CGFloat
-    private let maxY: CGFloat
-    private var polyhedronPosition: CGPoint = .zero
-    private var velocity: CGVector = .zero
+    private let polyhedraLayer: CALayer
+    private let cachedRenderings: ContiguousArray<CachedRendering>
+    private let radius: Int
+    private let maxX: Int
+    private let maxY: Int
+    private var positionX: Int = .zero
+    private var positionY: Int = .zero
+    private var velocityX: Int = .zero
+    private var velocityY: Int = .zero
     private var rotation: Int = .zero
 
     public init(size: CGSize, isPreview: Bool) {
         let settings = PolyhedraSettings()
-        self.polyhedraLayer = PolyhedraLayer(lineWidth: 1)
+        self.polyhedraLayer = CALayer()
+        self.polyhedraLayer.isOpaque = true
         self.radius = isPreview ? 25 : 150
-        self.velocity = isPreview ? CGVector(dx: 6, dy: 6) : CGVector(dx: 12, dy: 12)
+        self.velocityX = isPreview ? 6 : 12
+        self.velocityY = velocityX
         // make sure polyhedron (2 * radius in width) does not got off screen
-        self.maxX = size.width - 2 * radius
-        self.maxY = size.height - 2 * radius
+        self.maxX = Int(size.width) - 2 * radius
```

Comment:
Why does this need to be coerced into an int

Reply:
CGRect.width is a CGFloat so the cast is needed to go from a more precise to less precise. the [performance guide ](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CoreAnimation_guide/ImprovingAnimationPerformance/ImprovingAnimationPerformance.html#//apple_ref/doc/uid/TP40004514-CH9-SW1)says to make the coordinates all integral despite being stored as CGFloats